src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -68,6 +69,13 @@ internal sealed class VisualStudioProject
         private string? _outputRefFilePath;
         private string? _defaultNamespace;
 
+        /// <summary>
+        /// We store if this is a primary project or not as a tri-state.  Only <see langword="false"/> means that it's
+        /// not a primary project.  This way, if the value hasn't been set, or is never set, we still consider the
+        /// project to be a primary one.
```

Comment:
How is this different than just starting it out assigned as 'true' and removing the tri-state?

Reply:
that makes sense :)